see bottom of notebook for findings

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('../data/raw/TMDB_movie_dataset_v11.csv') #use nrows for faster loading 

In [37]:
df.head()

,id,title,vote_average,vote_count,status,release_date,revenue,runtime,adult,backdrop_path,...,original_title,overview,popularity,poster_path,tagline,genres,production_companies,production_countries,spoken_languages,keywords
0,27205,Inception,8.364,34495,Released,2010-07-15,825532764,148,False,/8ZTVqvKDQ8emSGUEMjsS4yHAwrp.jpg,...,Inception,"Cobb, a skilled thief who commits corporate es...",83.952,/oYuLEt3zVCKq57qu2F8dT7NIa6f.jpg,Your mind is the scene of the crime.,"Action, Science Fiction, Adventure","Legendary Pictures, Syncopy, Warner Bros. Pict...","United Kingdom, United States of America","English, French, Japanese, Swahili","rescue, mission, dream, airplane, paris, franc..."
1,157336,Interstellar,8.417,32571,Released,2014-11-05,701729206,169,False,/pbrkL804c8yAv3zBZR4QPEafpAR.jpg,...,Interstellar,The adventures of a group of explorers who mak...,140.241,/gEU2QniE6E77NI6lCU6MxlNBvIx.jpg,Mankind was born on Earth. It was never meant ...,"Adventure, Drama, Science Fiction","Legendary Pictures, Syncopy, Lynda Obst Produc...","United Kingdom, United States of America",English,"rescue, future, spacecraft, race against time,..."
2,155,The Dark Knight,8.512,30619,Released,2008-07-16,1004558444,152,False,/nMKdUUepR0i5zn0y1T4CsSB5chy.jpg,...,The Dark Knight,Batman raises the stakes in his war on crime. ...,130.643,/qJ2tW6WMUDux911r6m7haRef0WH.jpg,Welcome to a world without rules.,"Drama, Action, Crime, Thriller","DC Comics, Legendary Pictures, Syncopy, Isobel...","United Kingdom, United States of America","English, Mandarin","joker, sadism, chaos, secret identity, crime f..."
3,19995,Avatar,7.573,29815,Released,2009-12-15,2923706026,162,False,/vL5LR6WdxWPjLPFRLe133jXWsh5.jpg,...,Avatar,"In the 22nd century, a paraplegic Marine is di...",79.932,/kyeqWdyUXW608qlYkRqosgbbJyK.jpg,Enter the world of Pandora.,"Action, Adventure, Fantasy, Science Fiction","Dune Entertainment, Lightstorm Entertainment, ...","United States of America, United Kingdom","English, Spanish","future, society, culture clash, space travel, ..."
4,24428,The Avengers,7.710,29166,Released,2012-04-25,1518815515,143,False,/9BBTo63ANSmhC4e6r62OJFuK2GL.jpg,...,The Avengers,When an unexpected enemy emerges and threatens...,98.082,/RYMX2wcKCBAr24UyPD7xwmjaTn.jpg,Some assembly required.,"Science Fiction, Action, Adventure",Marvel Studios,United States of America,"English, Hindi, Russian","new york city, superhero, shield, based on com..."


In [39]:
df.shape

(1511625, 24)

In [40]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1511625 entries, 0 to 1511624
Data columns (total 24 columns):
 #   Column                Non-Null Count    Dtype  
---  ------                --------------    -----  
 0   id                    1511625 non-null  int64  
 1   title                 1511605 non-null  str    
 2   vote_average          1511625 non-null  float64
 3   vote_count            1511625 non-null  int64  
 4   status                1511625 non-null  str    
 5   release_date          1150231 non-null  str    
 6   revenue               1511625 non-null  int64  
 7   runtime               1511625 non-null  int64  
 8   adult                 1511625 non-null  bool   
 9   backdrop_path         367277 non-null   str    
 10  budget                1511625 non-null  int64  
 11  homepage              154451 non-null   str    
 12  imdb_id               689329 non-null   str    
 13  original_language     1511625 non-null  str    
 14  original_title        1511605 non-null  str  

In [48]:
df.columns

Index(['id', 'title', 'vote_average', 'vote_count', 'status', 'release_date',
       'revenue', 'runtime', 'adult', 'backdrop_path', 'budget', 'homepage',
       'imdb_id', 'original_language', 'original_title', 'overview',
       'popularity', 'poster_path', 'tagline', 'genres',
       'production_companies', 'production_countries', 'spoken_languages',
       'keywords'],
      dtype='str')

In [41]:
df.describe()

,id,vote_average,vote_count,revenue,runtime,budget,popularity
count,1.511625e+06,1.511625e+06,1.511625e+06,1.511625e+06,1.511625e+06,1.511625e+06,1.511625e+06
mean,9.506417e+05,1.473909e+00,1.419681e+01,6.027170e+05,4.295598e+01,2.287945e+05,9.505932e-01
std,5.047664e+05,2.808775e+00,2.763130e+02,1.973020e+07,6.035598e+01,5.263439e+06,6.598946e+00
min,2.000000e+00,0.000000e+00,0.000000e+00,-1.200000e+01,-2.800000e+01,0.000000e+00,0.000000e+00
25%,5.217850e+05,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00,1.430000e-02
50%,9.771610e+05,0.000000e+00,0.000000e+00,0.000000e+00,1.500000e+01,0.000000e+00,6.000000e-01
75%,1.389041e+06,0.000000e+00,0.000000e+00,0.000000e+00,8.400000e+01,0.000000e+00,6.550000e-01
max,1.791460e+06,1.000000e+01,3.449500e+04,5.000000e+09,1.440000e+04,1.000000e+09,2.994357e+03


In [42]:
df.isnull().sum()

id                            0
title                        20
vote_average                  0
vote_count                    0
status                        0
release_date             361394
revenue                       0
runtime                       0
adult                         0
backdrop_path           1144348
budget                        0
homepage                1357174
imdb_id                  822296
original_language             0
original_title               20
overview                 352554
popularity                    0
poster_path              554117
tagline                 1300121
genres                   680469
production_companies     879120
production_countries     749626
spoken_languages         719810
keywords                1145865
dtype: int64

In [69]:
df.isnull().sum().sort_values(ascending=False)

homepage                1357174
tagline                 1300121
keywords                1145865
backdrop_path           1144348
production_companies     879120
imdb_id                  822296
production_countries     749626
spoken_languages         719810
genres                   680469
poster_path              554117
release_date             361394
overview                 352554
title                        20
original_title               20
id                            0
vote_average                  0
runtime                       0
revenue                       0
status                        0
vote_count                    0
original_language             0
budget                        0
adult                         0
popularity                    0
dtype: int64

In [43]:
len(df)-len(df.drop_duplicates())

384

In [44]:
df.duplicated(subset='id', keep='first').sum()

np.int64(1376)

In [45]:
df.duplicated(subset='title', keep='first').sum()

np.int64(231780)

In [46]:
df.nunique()

id                      1510249
title                   1279844
vote_average               5025
vote_count                 3598
status                        6
release_date              43920
revenue                   14749
runtime                     840
adult                         2
backdrop_path            363269
budget                     6864
homepage                 144128
imdb_id                  683920
original_language           179
original_title          1319318
overview                1116290
popularity                20228
poster_path              951054
tagline                  201631
genres                    15613
production_companies     245946
production_countries      11354
spoken_languages           7768
keywords                 212945
dtype: int64

In [32]:
duplicate_ids = df[df.duplicated(subset='id', keep=False)]

duplicate_ids.sort_values('id').head(20)

,id,title,vote_average,vote_count,status,release_date,revenue,runtime,adult,backdrop_path,...,original_title,overview,popularity,poster_path,tagline,genres,production_companies,production_countries,spoken_languages,keywords
469789,1192939,Soldiers from Eastern Europe 6,0.0,0,Released,NaN,0,0,True,NaN,...,Soldiers from Eastern Europe 6,NaN,0.000,NaN,NaN,NaN,NaN,NaN,NaN,NaN
469791,1192939,Soldiers from Eastern Europe 6,0.0,0,Released,2004-04-04,0,90,True,NaN,...,Soldiers from Eastern Europe 6,11 horny & hung soldiers in 5 hot scenes! You ...,0.000,/jbvHXhHTPz5qCPosdSrfSrLBt0R.jpg,NaN,NaN,Eagle Video,Czech Republic,NaN,NaN
469783,1192939,Soldiers from Eastern Europe 6,0.0,0,Released,2004-04-04,0,90,True,NaN,...,Soldiers from Eastern Europe 6,11 horny & hung soldiers in 5 hot scenes! You ...,0.000,/jbvHXhHTPz5qCPosdSrfSrLBt0R.jpg,NaN,NaN,Eagle Video,Czech Republic,NaN,NaN
469779,1192939,Soldiers from Eastern Europe 6,0.0,0,Released,NaN,0,0,True,NaN,...,Soldiers from Eastern Europe 6,NaN,0.000,NaN,NaN,NaN,NaN,NaN,NaN,NaN
469736,1192940,杀夫,0.0,0,Released,NaN,0,0,False,NaN,...,杀夫,NaN,0.600,NaN,NaN,NaN,NaN,NaN,NaN,NaN
469697,1192940,杀夫,0.0,0,Released,NaN,0,0,False,NaN,...,杀夫,NaN,0.000,NaN,NaN,NaN,NaN,NaN,NaN,NaN
469680,1192941,Ekda Yeun Tar Bagha,0.0,0,Post Production,2023-11-24,0,0,False,NaN,...,एकदा येऊन तर बघा,upcoming marathi film,1.072,/lUygnhegALrTVnuTRQNSYbT3ndY.jpg,YOU WOULDN'T GO BACK,Comedy,"Ideas The Entertainment Company, SR Enterprise...",India,Marathi,NaN
469684,1192941,एकदा येऊन तर बघा,0.0,0,Released,NaN,0,0,False,NaN,...,एकदा येऊन तर बघा,upcoming marathi film,0.000,NaN,NaN,NaN,NaN,NaN,NaN,NaN
469691,1192941,Ekda Yeun Tar Bagha,0.0,0,Post Production,2023-11-24,0,0,False,NaN,...,एकदा येऊन तर बघा,upcoming marathi film,1.072,/lUygnhegALrTVnuTRQNSYbT3ndY.jpg,YOU WOULDN'T GO BACK,Comedy,"Ideas The Entertainment Company, SR Enterprise...",India,Marathi,NaN
469679,1192941,एकदा येऊन तर बघा,0.0,0,Released,NaN,0,0,False,NaN,...,एकदा येऊन तर बघा,upcoming marathi film,0.000,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [20]:
df['id'].value_counts().head(10)

id
1226029    4
1236071    4
1199990    4
1192952    4
1220680    4
1260267    4
1214170    4
1202079    4
1202467    4
1213633    4
Name: count, dtype: int64

In [21]:
df['imdb_id'].value_counts().head(10)

imdb_id
tt38023177    151
tt32094375    148
tt26900526     84
tt13904644     54
tt1757779      45
tt5719786      33
tt29703523     29
tt23810972     28
tt22075376     27
tt38990606     27
Name: count, dtype: int64

In [22]:
df['id'].value_counts().max()

np.int64(4)

In [23]:
df.loc[df['id']==1236071]

,id,title,vote_average,vote_count,status,release_date,revenue,runtime,adult,backdrop_path,...,original_title,overview,popularity,poster_path,tagline,genres,production_companies,production_countries,spoken_languages,keywords
328602,1236071,Semetua,9.0,1,Released,2018-02-07,0,44,False,NaN,...,Semetua,"Semetua was Sem Manutahi, master of ceremonies...",0.6,/yulre2ndAtvrLJ09u1R4wSKLtep.jpg,"Semetua was Sem Manutahi, master of ceremonies...",Documentary,In Vivo Prod,French Polynesia,French,NaN
328605,1236071,Semetua,9.0,1,Released,2018-02-07,0,44,False,NaN,...,Semetua,"Semetua was Sem Manutahi, master of ceremonies...",0.6,/yulre2ndAtvrLJ09u1R4wSKLtep.jpg,"Semetua was Sem Manutahi, master of ceremonies...",Documentary,In Vivo Prod,French Polynesia,French,NaN
590098,1236071,Semetua,0.0,0,Released,NaN,0,44,False,NaN,...,Semetua,"Semetua was Sem Manutahi, master of ceremonies...",0.0,NaN,"Semetua was Sem Manutahi, master of ceremonies...",NaN,NaN,NaN,NaN,NaN
590129,1236071,Semetua,0.0,0,Released,NaN,0,44,False,NaN,...,Semetua,"Semetua was Sem Manutahi, master of ceremonies...",0.0,NaN,"Semetua was Sem Manutahi, master of ceremonies...",NaN,NaN,NaN,NaN,NaN


In [12]:
(df['vote_count'] > 0).sum() # movies that have a valid vote count

np.int64(361176)

In [14]:
((df['budget'] > 0) & (df['revenue'] > 0)).sum() # movies that have both a valid budget and revenue

np.int64(18509)

In [28]:
(df['revenue'] <0).sum() # movies that have negative revenue

np.int64(1)

In [29]:
(df['revenue'] == 0).sum() # movies that have zero revenue

np.int64(1486059)

In [30]:
(df['runtime'] <0).sum() # movies that have negative runtime

np.int64(1)

In [31]:
(df['runtime'] == 0).sum() # movies that have zero runtime

np.int64(480695)

# Data Exploration Findings

**Dataset was downloaded** October 8, 2026. Findings are from that time. 

There are 1511625 (over 1.5 million) entries with 24 columns. The column names are as follows: 'id', 'title', 'vote_average', 'vote_count', 'status', 'release_date', 'revenue', 'runtime', 'adult', 'backdrop_path', 'budget', 'homepage', 'imdb_id', 'original_language', 'original_title', 'overview', 'popularity', 'poster_path', 'tagline', 'genres', 'production_companies', 'production_countries', 'spoken_languages', 'keywords'.

## Duplicate entries
There are 384 duplicate rows. There are 1376 rows with repeated IDs, excluding the first occurrence of each ID, and 231780 repeated titles, excluding the first occurrence of each title. The ids are worth investigating, but duplicate titles are nothing unusual. When counting duplicate ids, we saw that there are at most 4 entries with the same id. An example is id=1236071 which shows four different entries for the movie "Semetua". The first 2 seem to be identical, the other 2 do have the same title but are mostly empty for other columns. These problems need to be tackled during cleaning.

## Missing values
Many movies are missing homepage, tagline, keywords, backdrop_path, production_companies, imdb_id, production_countries and spoken_languages. Importance of these columns depends on analysis. A more interesting column is genres, which is missing for 680469 movies. poster_path, release_date and overview are also missing for a lot of movies. title and original_title are missing for 20 entries. 1 movie has negative revenue, 1486059 movies have revenue 0. There are only 18509 movies that have both positive revenue and budget. 1 movie has negative runtime, 480695 have runtime 0. Only 361176 movies have at least one vote.